src/index.js

```diff
@@ -1 +1,61 @@
+/* eslint-disable no-console */
 'use strict';
+
+const http = require('http');
+const fs = require('fs');
+const path = require('path');
+const querystring = require('querystring');
+
+const server = new http.Server();
+
+const PORT = process.env.PORT || 3000;
+
+server.on('request', (req, res) => {
+  if (req.method === 'GET') {
+    const fileName = path.join(__dirname, 'index.html');
+    const file = fs.ReadStream(fileName);
+
+    file.on('error', () => {
+      res.statusCode = 404;
+      res.end('File not found');
+    });
+
+    file.pipe(res);
+  } else if (req.method === 'POST') {
+    let body = '';
+
+    req.on('data', (chunk) => {
+      body += chunk.toString();
+    });
+
+    req.on('end', () => {
+      const formData = querystring.parse(body);
+      const date = formData.date;
+      const title = formData.title;
+      const amount = formData.amount;
+
+      const formDataObject = {
+        date,
+        title,
+        amount,
+      };
+
+      const jsonData = JSON.stringify(formDataObject);
+
+      fs.writeFileSync('formData.json', jsonData);
+      res.setHeader('Content-Type', 'text/html');
+      res.statusCode = 200;
+      res.end(jsonData);
+    });
```

Comment:
move the handlers to separate functions to improve readability.
make your server look like here:
<img width="440" alt="image" src="https://github.com/mate-academy/node_form-data/assets/98764641/a2a97d4b-43af-41c0-9466-b176b2374004">